helix-core/src/selection.rs

```diff
@@ -497,6 +497,12 @@ impl Selection {
     }
 }
 
+impl From<Range> for Selection {
+    fn from(range: Range) -> Self {
+        Self::single(range.anchor, range.head)
+    }
+}
```

Comment:
cc @cessen Is this a good idea? I think `Selection::single` is more explicit but this seemed to work as well, maybe later we might do `Vec<Range>` too.

Reply:
I don't feel strongly about it, but I agree that this seems redundant with `Selection::single()`.  So I think we should get rid of one or the other of them.

I think I lean slightly towards keeping `Selection::single()`, because my general experience with the `From/Into` traits is that you can sometimes end up in situations where the type system can't figure out the intended type, even though it seems like it should be able to.  And without type ascription (which is still unstable) that forces you to break things out into separate variables just to be able to explicitly specify their type.

Reply:
Agreed, let's remove this implementation, maybe we could change `Selection::single` to take a `Range` instead of two usizes.

Reply:
Removed the `From` reverted to using `single()`.